File: Sources/Localization/SupportedLocale.swift
import Foundation

public enum SupportedLocale: String, CaseIterable, Codable, Equatable, Hashable, Sendable {
    case enUS = "en-US"
    case es = "es"
    case ptBR = "pt-BR"
    case ja = "ja"
    case de = "de"
    case fr = "fr"
    case he = "he"
    case da = "da"
    case fi = "fi"
    case it = "it"
    case nl = "nl"
    case nb = "nb"
    case sv = "sv"

    public static let sourceLocale: SupportedLocale = .enUS

    public static let targetLocales: [SupportedLocale] = [
        .es,
        .ptBR,
        .ja,
        .de,
        .fr,
        .he,
        .da,
        .fi,
        .it,
        .nl,
        .nb,
        .sv
    ]

    public static var supportedLocales: [SupportedLocale] {
        [sourceLocale] + targetLocales
    }

    public static var supportedLocaleIdentifiers: [String] {
        supportedLocales.map(\.identifier)
    }

    public var identifier: String {
        rawValue
    }

    public var locale: Locale {
        Locale(identifier: rawValue)
    }
}

File: src/main/kotlin/com/vadimkomis/localization/SupportedLocale.kt
package com.vadimkomis.localization

import java.util.Locale

enum class SupportedLocale(val identifier: String) {
    EN_US("en-US"),
    ES("es"),
    PT_BR("pt-BR"),
    JA("ja"),
    DE("de"),
    FR("fr"),
    HE("he"),
    DA("da"),
    FI("fi"),
    IT("it"),
    NL("nl"),
    NB("nb"),
    SV("sv");

    val locale: Locale
        get() = Locale.forLanguageTag(identifier)

    companion object {
        val sourceLocale: SupportedLocale = EN_US

        val targetLocales: List<SupportedLocale> = listOf(
            ES,
            PT_BR,
            JA,
            DE,
            FR,
            HE,
            DA,
            FI,
            IT,
            NL,
            NB,
            SV
        )

        val supportedLocales: List<SupportedLocale> = listOf(sourceLocale) + targetLocales

        val supportedLocaleIdentifiers: List<String> = supportedLocales.map { it.identifier }

        fun fromIdentifier(identifier: String): SupportedLocale? =
            entries.firstOrNull { it.identifier == identifier }
    }
}
